src/masm/x64.rs

```diff
@@ -521,6 +521,28 @@ impl MacroAssembler {
         asm::cvtsd2ss(self, dest, src);
     }
 
+    pub fn int_as_float(
+        &mut self,
+        dest_mode: MachineMode,
+        dest: FReg,
+        src_mode: MachineMode,
+        src: Reg,
+    ) {
+        asm::pxor(self, dest, dest);
```

Comment:
I suppose we don't need this all the time, clang for example doesn't emit this [here](https://gcc.godbolt.org/z/BtMbDb). I guess we could make it conditional like `if dest_mode.size() < src_mode.size() { asm::pxor(...) }`.

Reply:
Will do, thanks!

Reply:
I think we don't need it at all for the `as*` family of functions, as they only operate on src_mode.size() == dest_mope.size()`.